ampled-music/ampled
# (387) Allow artist page admin to delete their artist page

https://trello.com/c/etMmJj9H/387-delete-artist-page

This PR introduces the feature to allow artist page admins to delete their artist page.  Once the artist page is deleted, subscriptions and posts will be deleted as well.

Testing:
- Go to `/create-artist`
  - Before saving the new page, the Delete your page button should not be visible
  - After saving the page, test both admin and non-admin users for the page:
    - The user who created the page (who will become admin by default) and any other designated admins should see the Delete your page button at the bottom of the screen
    - Add a member to the artist page who is not an admin; login as that member and view the artist page in edit mode; you should not see the Delete your page button
- Setting up for testing after deletion:
  - Create a few posts for the artist page
  - Create another user and have subscribe to the artist page 
- On Delete your page click
  - A modal confirmation popup appears confirming that you're sure you want to delete your page (if you have any subscribed supporters, the modal will show you your supporter count)
    - Clicking Cancel will not alter anything and close the modal, bringing you back to edit mode
    - Clicking to confirm will initiate the deletion process; upon success:
      - You should see a toast notification telling you that the deletion is successful
      - You should be redirected back to the main landing page
      - Any created posts will no longer be available (and the links won't work)
      - Subscriptions will be cancelled (subscribed users will get the email notification that their subscription has been cancelled; currently this is using the same email as the one users who initiate their cancellation receive)
- Deleting the artist page as a site admin (not as the page admin) should not cancel any open subscriptions

TODO: 
- Delete any audio files associated w/ artist page posts [moving to separate PR as this feels outside of scope, this should happen whenever a post is deleted not just when deleted as a result of artist page deletion] (oh yeah there's already a Trello for it! https://trello.com/c/znJ9DxZ3/553-images-never-get-deleted-from-cloudinary-even-if-associated-posts-or-pages-or-profiles-get-removed)
